lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -5673,6 +5725,68 @@ bool ReplaceZeroScaleFP16QuantNodes::run(Function *F,
   return changed;
 }
 
+/// If BatchNorm type is FP32 and preceding Conv is FP16, convert BatchNorm
+/// precision to FP16, to allow BatchNorm to fuse into Convolution
+bool ConvertBatchNormPrecision::run(Function *F,
+                                    const CompilationContext &cctx) {
+  LOG_SCOPE(F->getLogContext(), getName());
+  bool changed = false;
+  for (auto &node : F->getNodes()) {
+    if (auto *BN = dyn_cast<BatchNormalizationNode>(&node)) {
+      if (auto *CT = dyn_cast<ConvertToNode>(BN->getInput())) {
```

Comment:
```suggestion
    auto *BN = dyn_cast<BatchNormalizationNode>(&node);
    if (!BN) {
      continue;
    }
    auto *CT = dyn_cast<ConvertToNode>(BN->getInput());
    if (!CT) {
      continue;
    }
```

Reply:
Maybe also check CT is converting to FP32?

Reply:
Added the check